### DISCUSSION TITLE
make jtx board events show up in evolution; option to use VEVENT

### DISCUSSION THREAD

**Discussion by justpeanuts** [TechbeeAT/jtxBoard#836] [Ideas]
As far as I understand, the journal entries made by jtx board don't show up in other regular calendar apps like evolution or simple calendar - this is because jtx board use VJOURNAL not VEVENT. VJOURNAL is not supported by regular calendar apps. (Please correct me if I am wrong) 
If so, can you please make an option to use regular VEVENT for making journal entries instead of VJOURNAL ? This would make it compatible with existing calendar apps.

Edit, PS : Forgot to mention how excellent an app this is! I tried it some time ago but then it only had a few features but now it is a super well made app and has features at par and beyond all available task and calendar apps.. Thank you for making such a nice piece of free and open source software for humanity! By the only problem is I don't know how to make it work with existing calendar apps and access its data.

**Comment by rogercreagh** [comment]
I expect @patrickunterwegs will give a fuller response but just to say that Evolution does handle VJournal items
Here's a screenshot.
![Screenshot from 2023-08-28 15-50-54 resized](https://github.com/TechbeeAT/jtxBoard/assets/44980188/d6393cc3-46fd-46bb-9694-40373ba6c814)
Its not as good as jtxB, but it does work - they call them Memos and they put the undated ones (which jtxB separates as 'Notes') at the top of the list.

Unlike a calendar which is optimised for VEvents (which have to have a start and end time, and typically are intended for scheduling future events - Calendars often auto-delete events more than a month old by default to save space and speed things up) jtxB is optimised for VJournals and is brilliant for keeping a journal or diary of past events.

Most Calendars and jtxB also handle VTodo items, as do dedicated Task apps (which typically don't handle Vevents or VJournals) 

All three types of item can be stored in a CalDav calendar.

Personally I would not see the point of jtxB handling VEvents - there are many good calendar apps out there already, and a Journal is not the same as a Calendar. There may be space for a VEvent app which handled events more like jtxB handles journals/notes, but that would be a different thing.

**Comment by patrickunterwegs** [comment]
Thanks @rogercreagh , you already said most of the relevant points. 

I have two additions: 

1) Basically it would be easy to convert a VJOURNAL into a VEVENT. The issue is that Android has a local calendar storage that can be accessed by any calendar app. But this local calendar storage only supports VEVENTs. That's why apps implementing VJOURNALs or VTODOs need to come up with their own data storage (which is usually a database within the app).
Now when a VJOURNAL should be saved as a VEVENT, it only makes sense when it's stored in the local calendar storage. But the technical storage is a completely different one. 
But apart from the technical effort it would also just abuse the VEVENT component as the VJOURNAL component was defined for its own purpose anyway.

2) For the future I consider a calendar integration, but for a start only to show the calendar entries in the local calendar storage (probably just read-only). The idea is to easily link your entries to a calendar entry (or e.g. to create a task as a subtask to an event). But that's for the future. I wouldn't consider saving VJOURNALs as VEVENTs though.

**Reply by bepolymathe** [reply]
Hi, 

> For the future I consider a calendar integration

That’s a great news !

**Comment by justpeanuts** [comment]
Thank you for taking the time out to give a detailed reply! @rogercreagh @patrickunterwegs 
If I understand correctly using VJOURNAL (and utilising features that it offers in addition to that offered by VEVENT) is key part of jtxB that makes it unique w.r.t other calendar apps. My query was mainly there because it felt like the data that is put into jtxB was stuck inside there because, within android, there is no way for other apps to access the data inside jtxB and there is no way for jtxB to access (say link to) notes in other apps. This might be topic for another discussion. I mainly use plain text note apps - the only way to convert jtxB to plain text would be sync the data with Davx5 to a computer and then use scripts inside the computer by querying the data in the calendar server software to access the data and make it onto plain text form. But I don't have an always on server that keeps doing it.

Use cases for this can include may be making a script (using termux etc.) for doing something if there is a note with a particular content (eg. Meeting with Bob) do something instantly (eg. get a reminder with notes that were made to remember for the meeting with Bob ) or maybe to add notes programmatically to jtxB from within android. I understand such use cases are all not very practical - only the point that I am making is that there should be ways to make the data in the one app with instantly usable in other apps and for jtxB to be able to use data in other apps.

**Reply by patrickunterwegs** [reply]
Hey @justpeanuts , 
I get your point that the format is currently limited to be used in jtx Board. But you have to understand that the idea of using the iCalendar Standard and VJOURNAL component was and is the very opposite. 
Most other apps use proprietary formats that you can't use in any other app because you might not even know which format is used in the background. In the best case there are just .md files.

I made jtx Board for exactly the reason to strengthen the iCalendar Standard, to ensure interoperability and maybe inspire other developers to come up with other solutions (e.g. with a web app). 

Sure I understand that interoperability is limited as there are just very limited ways to show the entries in other apps, but I would still like to kind of insist on using the standard workout workarounds and hoping that others would still get inspired and make something similar! 😉
